01/app.js

```diff
@@ -0,0 +1,13 @@
+const form = document.querySelector("form");
```

Comment:
warto pamiętać o `DOMContentLoaded`

Reply:
Widzialam uzycie tego w zadaniu 04 jednak nie do konca rozumiem dlaczego to stosujemy w tym przypadku. Tak jak to rozumiem listener ten jest potrzebny zeby najpierw zaladowal sie content HTML i zeby skrypt nie zostal odpalony zanim to sie stanie. Umieszczanie skryptu na dole strony powinno temu zapobiegac? Alternatywnie uzycie tagu defer w skrypcie jesli jest umieszczony na gorze strony. Czy to ma jakies dodatkowe funkcje jeszcze? I jak to zastosowac, widzialam zastosowanie w zadaniu 04 i problemem, ktory napotkalam jest to ze znowu boxElement z funckji init() byl dla mnie niedostepny. Po raz kolejny go wybralam w linii 54. Tak samo probujac dodac tutaj listener dla DOMContentLoad i wybierajac forme do zmiennej nie moge sie do niej dostac pozniej by uzyc na niej event listenera, a jesli wszystko umieszcze w srodku tej event listenera dla DOMContentLoad to nic nie dziala? :)

Reply:
`DOMContentLoaded` to zabezpieczenie dla nas, że DOM jest załadowany. Nie zawsze mamy możliwość decyzji gdzie plik js (czy w `head` czy pod koniec `body`) jest podpięty. Jak użyjemy tego eventu to mamy pewność, że zawsze będzie działać.
Wtedy wyszukiwania robimy wew. tego eventu natomiast funkcje, które operują na wyszukanych elementach otrzymują je 
przez parametr i wtedy nie będzie problemu z zasięgiem (wszystko powinno działać).

Reply:
Dzieki, rozumiem :) Jednak dalej nie rozumiem jak mam sie dostac do zmiennej form jesli deklaruje ja w funkcji. Czy mam ja zwyczajnie zadeklarowac jeszcze raz poza funkcja? Rozpisalam kod z DOMContentLoad i dalam w kodzie caly komentarz.